scalding-hadoop-test/src/main/scala/com/twitter/scalding/platform/HadoopPlatform.scala

```diff
@@ -33,6 +35,9 @@ trait HadoopPlatform[P, R, T <: HadoopPlatform[P, R, T]] {
 
   def run(): Unit
 
+  def runAndHandle(fn: Throwable => Unit): Unit =
+    fn(Try { run() }.failed.get)
```

Comment:
So, this method needs when you expect an exception to check exception path, in this case success unexpected behaviour and user should catch it as exception.

not sure about type parameter, but let's do it.